dolphin/src/logger/layer.rs

```diff
@@ -153,34 +153,16 @@ impl DolphinLoggerVisitor {
         let level = metadata.level();
 
         // Dolphin logs in the format of {Minutes}:{Seconds}:{Milliseconds}.
-        let time = OffsetDateTime::now_local().unwrap_or_else(|e| {
-            eprintln!("[dolphin_logger/layer.rs] Failed to get local time: {:?}", e);
-
-            // This will only happen if, for whatever reason, the timezone offset
-            // on the current system cannot be determined. Frankly there's bigger issues
-            // than logging if that's the case.
-            OffsetDateTime::now_utc()
-        });
-
-        let mins = time.minute();
-        let secs = time.second();
-        let millsecs = time.millisecond();
-
-        // We want 0-padded mins/secs, but we don't need the entire formatting infra
-        // that time would use - and this is simple enough to just do in a few lines.
-        let mp = match mins < 10 {
-            true => "0",
-            false => "",
-        };
+        // Get local time with proper timezone handling
+        let local_time: DateTime<Local> = Local::now();
+        let time: DateTime<FixedOffset> = local_time.with_timezone(local_time.offset());
 
-        let sp = match secs < 10 {
-            true => "0",
-            false => "",
-        };
+        // Extract milliseconds for Dolphin's log format (chrono doesn't have milliseconds API)
+        let milliseconds: u32 = time.nanosecond() / 1_000_000;
+
+        let formatted_time_string: String = time.format("%M:%S").to_string() + &format!(":{}", milliseconds);
 
-        Self(format!(
-            "{mp}{mins}:{sp}{secs}:{millsecs} {file}:{line} {level}[{log_type}]: "
-        ))
+        Self(format!("{} {file}:{line} {level}[{log_type}]: ", formatted_time_string))
```

Comment:
I get the feeling this might be better as something like:

``` rust
Self(format!("{}:{} {file}:{line} {level}[{log_type}]: ", time.format("%M:%S"), milliseconds))
```

Otherwise this is doing more `String` allocations than it needs to; `.format()` on `time` should be a `DelayedFormat`, which should impl `Display` and "just work" here.

(I'm open to being wrong, but it's the logging path which could get called a lot)